cpp/src/text/subword/wordpiece_tokenizer.cu

```diff
@@ -520,11 +520,19 @@ void wordpiece_tokenizer::tokenize(uvector_pair& cps_and_offsets, rmm::cuda_stre
   // Repurpose the input array for the token ids. In the worst case, each code point ends up being a
   // token so this will always have enough memory to store the contiguous tokens.
   uint32_t* contiguous_token_ids = device_code_points;
-  thrust::copy_if(rmm::exec_policy(stream),
-                  device_token_ids.begin(),
-                  device_token_ids.end(),
-                  contiguous_token_ids,
-                  copy_if_fn{});
+  auto const copy_size           =  // thrust::copy_if limited to copying int-max values
+    std::min(device_token_ids.size(), static_cast<std::size_t>(std::numeric_limits<int>::max()));
+  auto ids_itr       = device_token_ids.begin();
+  auto const ids_end = device_token_ids.end();
+  while (ids_itr != ids_end) {
+    auto const copy_end = static_cast<std::size_t>(std::distance(ids_itr, ids_end)) <= copy_size
```

Comment:
My personal preference for ternaries (and in general) to avoid operator precedence issues. You can reject if you strongly dislike the extra parens though.
```suggestion
    auto const copy_end = (static_cast<std::size_t>(std::distance(ids_itr, ids_end)) <= copy_size)
```